frontend/test/pytest/test_gradient.py

```diff
@@ -688,5 +688,44 @@ def interpret_grad(params):
     assert np.allclose(compile_grad(jnp.array(inp)), interpret_grad(inp))
 
 
+@pytest.mark.parametrize("diff_method", ["fd", "adj", "ps"])
+@pytest.mark.parametrize("inp", [(1.0), (2.0), (3.0), (4.0)])
+def test_finite_diff_multiple_devices(inp, diff_method, backend):
+    """Test gradient methods using multiple backend devices."""
+
+    @qml.qnode(qml.device(backend, wires=1))
+    def f(x):
+        qml.RX(3 * x, wires=0)
+        return qml.expval(qml.PauliY(0))
+
+    @qml.qnode(qml.device("lightning.qubit", wires=1))
+    def g(x):
+        qml.RX(3 * x, wires=0)
+        return qml.expval(qml.PauliY(0))
+
+    @qjit()
+    def compiled_grad_default(params, ntrials):
+        d_f = grad(f, argnum=0, method=diff_method)
+
+        def fn_f(i, g):  # pylint: disable=unused-argument
+            return d_f(params)
+
+        d_g = grad(g, argnum=0, method=diff_method)
+
+        def fn_g(i, g):  # pylint: disable=unused-argument
```

Comment:
```suggestion
        def fn_g(_, g):
```

Reply:
The issue remains for unused variable `g` :(

Reply:
It works for any number of arguments, just give them a leading underscore 👍